CMSgov/mint-app
# [EASI-4386] - Homepage customization settings

# EASI-4386


## Changes and Description

- Added view for selecting homepage settings
- Added view for selecting order of settings
- Added view for selection operational solutions
- Unit tests
- Cypress tests
- Feature flag around routes
- Success message

Data only persists on the solutions page (to save solutions) as well as the order page.  Selecting the settings on the settings page stores data in FE memory/router state.  Verify no mutation occurs on the settings page when you click 'Next', only on the order page when you select 'Save customizations'.



## How to test this change

- Navigate to `/homepage-setttings`
- Verify that `My Model Plans` is selected by default if base user
- Verify that `My Model Plans` (if collaborator) and `All Model Plans` selected is admin
- Verify that `Models with FFS CRs or TDLs` selected if FFS/Mac user
- Click selections, and select `Select solutions` on the relevant card
- In the dropdown, choose some solutions and save.  
- Verify selections persisted, solutions rendered on card.
- Make changes to setting selections if you wish and click next
- Change order, click back, forward, verify state persists.
- Click 'Save customizations' on order page, verify success message
- Go back and verify selections persist

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
